internal/yunikorn/sync.go

```diff
@@ -143,24 +141,6 @@ func (s *Service) upsertPartitionQueues(ctx context.Context, partitions []*dao.P
 	return queues, nil
 }
 
-// flattenQueues returns a list of all queues in the hierarchy in a flat array.
-// Usually the returned queues are a single hierarchical structure, the root queue,
-// and all other queues are children queues.
-func flattenQueues(qs []*dao.PartitionQueueDAOInfo) []*dao.PartitionQueueDAOInfo {
-	var queues []*dao.PartitionQueueDAOInfo
-	for _, q := range qs {
-		queues = append(queues, q)
-		if len(q.Children) > 0 {
-			// update partitionName for children #148
-			for i := range q.Children {
-				q.Children[i].Partition = q.Partition
-			}
-			queues = append(queues, flattenQueues(util.ToPtrSlice(q.Children))...)
-		}
-	}
-	return queues
-}
-
```

Comment:
We need to revert it for now, as we are not modifying `UpsertQueue()` anymore. In later ticket we have to update the sync logic and replace `UpsertQueue()` with AddQueues() and UpdateQueues().